test/xpu_api/tuple/tuple.tuple/moveable2_g.pass.cpp

```diff
@@ -0,0 +1,82 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/tuple>
+#include <oneapi/dpl/utility>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+struct MoveOnly
+{
+    MoveOnly() {}
+
+    MoveOnly(MoveOnly&&) {}
+
+    MoveOnly&
+    operator=(MoveOnly&&)
+    {
+        return *this;
+    }
+
+    MoveOnly(MoveOnly const&) = delete;
+    MoveOnly&
+    operator=(MoveOnly const&) = delete;
+};
+
+MoveOnly
+make_move_only()
+{
+    return MoveOnly();
+}
```

Comment:
For me it's look ok: we simple checking that able to return from the function inside Kernel something available only to move.

Reply:
Even if this this function was intended to test something like this, it is not the purpose of this test. 
And also since C++17, such return from the function is guaranteed to be optimized-out so passing `make_move_only()` to the tested function is equivalent to directrly call the default constructor of MoveOnly().

Reply:
Done